controller/node_controller.go

```diff
@@ -1120,6 +1121,43 @@ func (nc *NodeController) syncMultipathd(node *longhorn.Node, namespaces []lhtyp
 	node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeMultipathd, longhorn.ConditionStatusTrue, "", "")
 }
 
+func (nc *NodeController) checkModulesLoaded(node *longhorn.Node, namespaces []lhtypes.Namespace) {
+	checkingModules := []string{
+		"dm_crypt",
+	}
+	nsexec, err := lhns.NewNamespaceExecutor(lhtypes.ProcessNone, lhtypes.HostProcDirectory, namespaces)
+	if err != nil {
+		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeModulesLoaded, longhorn.ConditionStatusFalse,
+			string(longhorn.NodeConditionReasonNamespaceExecutorErr),
+			fmt.Sprintf("Failed to get namespace executor: %v", err.Error()))
+		return
+	}
+
+	lsModInfo, err := nsexec.Execute(nil, "lsmod", nil, lhtypes.ExecuteDefaultTimeout)
+	if err != nil {
+		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeModulesLoaded, longhorn.ConditionStatusFalse,
+			string(longhorn.NodeConditionReasonNamespaceExecutorErr),
+			fmt.Sprintf("Failed to get lsmod information: %v", err.Error()))
+		return
+	}
+
+	notLoadedModules := []string{}
+	for _, module := range checkingModules {
+		if strings.Contains(lsModInfo, module) {
+			continue
+		}
+		notLoadedModules = append(notLoadedModules, module)
+	}
+	if len(notLoadedModules) != 0 {
+		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeModulesLoaded, longhorn.ConditionStatusFalse,
+			string(longhorn.NodeConditionReasonModulesNotLoaded),
+			fmt.Sprintf("Modules %v are not loaded on node %v", notLoadedModules, node.Name))
+		return
+	}
```

Comment:
No, it is not listed from the `lsmod` command.
It will need to inspect the kernel configuration `/boot/config-$(uname -r)` as what we do for checking the NFS client version.

Reply:
Sounds good

Reply:
Modified. Thanks